code/modules/mapping/ruins.dm

```diff
@@ -115,7 +115,7 @@
 		if(R.unpickable)
 			continue
 		ruins_available[R] = R.placement_weight
-	while((budget > 0 || mineral_budget > 0) && (ruins_available.len || forced_ruins.len))
```

Comment:
The CI test doesn't touch the budget. It sets the ruin cost per ruin to 0. So if your budget is say, 16, it never decreases because the cost of all ruins is 0.

Reply:
I can't believe I have to post this but:

https://github.com/tgstation/tgstation/blob/45c4edc57962406d4a7e11ec76b62884b5ff7900/tools/ci/ci_config.txt#L6

Reply:
Interesting. I was looking at the wrong spot then:

https://github.com/tgstation/tgstation/blob/45c4edc57962406d4a7e11ec76b62884b5ff7900/code/controllers/configuration/entries/game_options.dm#L345-L358

Reply:
Why didn't you just delete the space and lavaland budget from the config?

Reply:
> Why didn't you just delete the space and lavaland budget from the ci_config?

I originally didn't intend on fixing the CI test, I was fixing a bug where the entire always_spawn_with system didn't work because when the budget ran out, it couldn't spawn the ruin forced by always_spawn_with. 

The ruins are also forced. It doesn't make sense for something that is forced to respect the budget.